### DISCUSSION TITLE
Add Storage::link() support

### DISCUSSION THREAD

**Discussion by MarcBresson** [laravel/framework#41470] [Ideas]
Hello,

I encountered a case where `Storage::link($target, $source)` would be really handy. It would just create a symlink from target to source.

I tried to implement it myself with the use of `Illuminate\Filesystem\Filesystem::link()` function, but `FileSystem` keeps getting deeper. I eventually reach all the files involved but it didn t work (without throwing exception).  I am no expert to handle this.

**Comment by chuckrincon** [comment]
Hi there ✋🏻 

From version 7.x and up, you may configure additional symbolic links in your filesystems configuration file.
Each of the configured links will be created when you run the `php artisan storage:link` command:

e.g. This will create a symlink from target (public) to source (storage)
```php
'links' => [
    public_path('invoices') => storage_path('app/invoices'),
    public_path('products') => storage_path('app/products'),
    public_path('avatars') => storage_path('app/avatars'),
]
```

Once a file has been stored and the symbolic link has been created, you can create a URL to the files using the asset helper 
e.g.
```php
asset("invoices/$invoice->uuid");
asset("products/$product->cover");
asset("avatars/$user->avatar");
```

**Comment by MarcBresson** [comment]
Hello again :)

thanks for your answer. However, I was more speaking of a symlink between files inside the storage section. For example, if i want logo.png and logo-big.png to be the same, I would only need a symlink instead of duplicating the picture !

Hope I made what i wanted clear :)
